src/Scripts/Python/RekoProgramAPI.cs

```diff
@@ -160,71 +169,78 @@ public void SetUserProcedure(ulong linearAddress, string decl)
             }
         }
 
-        public bool GetProcedureDecompileFlag(ulong linearAddress)
+        public bool GetProcedureDecompileFlag(Address addr)
         {
-            return UserProcedure(linearAddress).Decompile;
+            return UserProcedure(addr).Decompile;
         }
 
         public void SetUserProcedureDecompileFlag(
-            ulong linearAddress, bool decompile)
+            Address addr, bool decompile)
         {
-            UserProcedure(linearAddress).Decompile = decompile;
+            EnsureUserProcedure(addr).Decompile = decompile;
         }
 
-        public string? GetProcedureOutputFile(ulong linearAddress)
+        public string? GetProcedureOutputFile(Address addr)
         {
-            return UserProcedure(linearAddress).OutputFile;
+            return UserProcedure(addr).OutputFile;
         }
 
         public void SetUserProcedureOutputFile(
-            ulong linearAddress, string outputFile)
+            Address addr, string outputFile)
         {
-            UserProcedure(linearAddress).OutputFile = outputFile;
+            EnsureUserProcedure(addr).OutputFile = outputFile;
         }
 
         private bool TryConvertAddress(
-            ulong linearAddress, [NotNullWhen(true)] out Address? addr)
+            string sAddress, [NotNullWhen(true)] out Address? addr)
         {
-            try
-            {
-                addr = Addr(linearAddress);
-                return true;
-            }
-            catch
-            {
-                addr = null;
-                return false;
-            }
-        }
-
-        private Address Addr(ulong linearAddress)
-        {
-            return program.SegmentMap.MapLinearAddressToAddress(linearAddress);
+            return program.Architecture.TryParseAddress(sAddress, out addr);
         }
 
-        private EndianImageReader CreateImageReader(ulong linearAddress)
+        private EndianImageReader CreateImageReader(Address addr)
         {
-            var addr = Addr(linearAddress);
             return program.CreateImageReader(program.Architecture, addr);
         }
 
         private IEnumerable<T> ReadData<T>(
             Func<EndianImageReader, T> reader,
-            ulong iStartAddr,
+            Address startAddr,
             long length)
         {
-            var rdr = CreateImageReader(iStartAddr);
+            var rdr = CreateImageReader(startAddr);
             var offStart = rdr.Offset;
             while (rdr.Offset - offStart < length)
             {
                 yield return reader(rdr);
             }
         }
 
-        private UserProcedure UserProcedure(ulong linearAddress)
+        private UserProcedure UserProcedure(Address addr)
+        {
+            TryGetUserProcedure(addr, out var userProc);
```

Comment:
Isn't this a bit of a lie? True, we have a userproc, but it's not registered in the `program`.

Reply:
Agree. I tried to make it clear at 2408f40